src/main/java/com/sap/oss/phosphor/fosstars/model/value/ArtifactVersion.java

```diff
@@ -0,0 +1,153 @@
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+/**
+ * This class represents a specific version of an artifact that is produced by an open source
+ * project. For example, it may be a jar file.
+ */
+public class ArtifactVersion {
+
+  public static final ArtifactVersion EMPTY = new ArtifactVersion("", LocalDate.now());
+
+  /**
+   * Comparator for artifact versions release date.
+   */
+  private static final Comparator<ArtifactVersion> RELEASE_DATE_COMPARISON =
+      (a, b) -> b.getReleaseDate().compareTo(a.getReleaseDate());
+
+  private final String version;
+
+  @JsonDeserialize(using = LocalDateDeserializer.class)
+  @JsonSerialize(using = LocalDateSerializer.class)
+  private final LocalDate releaseDate;
+
+  @JsonIgnore
+  private final Optional<SemanticVersion> semanticVersion;
+
+  /**
+   * Initialize the ArtifactVersion based on version tag and release date.
+   *
+   * @param version version tag
+   * @param releaseDate release date
+   */
+  public ArtifactVersion(@JsonProperty("version") String version,
+      @JsonProperty("releaseDate") LocalDate releaseDate) {
+
+    Objects.requireNonNull(version, "Version must be set");
+    Objects.requireNonNull(releaseDate, "Release date must be set");
+
+    this.version = version;
+    this.releaseDate = releaseDate;
+    this.semanticVersion = SemanticVersion.parse(version);
+  }
+
+  /**
+   * Sort artifact versions by release date.
+   *
+   * @param versions the artifact versions
+   * @return sorted collection of ArtifactVersion
+   */
+  public static Collection<ArtifactVersion> sortByReleaseDate(Set<ArtifactVersion> versions) {
+    SortedSet<ArtifactVersion> sortedArtifacts = new TreeSet<>(RELEASE_DATE_COMPARISON);
+    sortedArtifacts.addAll(versions);
+    return sortedArtifacts;
+  }
+
+  public Optional<SemanticVersion> getSemanticVersion() {
+    return semanticVersion;
+  }
+
+  @JsonIgnore
+  public boolean isValidSemanticVersion() {
+    return semanticVersion.isPresent();
+  }
+
+  public String getVersion() {
+    return version;
+  }
+
+  public LocalDate getReleaseDate() {
+    return releaseDate;
+  }
+
+  @Override
+  public String toString() {
+    return "{"
+        + "version='" + version + '\''
+        + ", releaseDate=" + releaseDate
+        + '}';
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    ArtifactVersion compareVersion = (ArtifactVersion) o;
+    return version.equals(compareVersion.version) && releaseDate.equals(compareVersion.releaseDate);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(version, releaseDate);
+  }
+
+  /**
+   * LocalDate to Date deserializer used by Jackson Databind for JSON parsing.
+   */
+  private static class LocalDateDeserializer extends StdDeserializer<LocalDate> {
```

Comment:
Have you tried something like `@JsonFormat(pattern = "yyyy-MM-dd")` instead?

Reply:
Yes, but this requires to add `jackson-datatype-jsr310` as dependency.
For me adding the de-/serializer was leaner compared to adding the dependency only for this.

However, based on our discussion in another comment, the `LocalDate` may be replaced by `Date` as there may be two releases at the same date but different time.
I like `LocalDate` more but I have not considered multiple releases per day 😕
Have updated #478 accordingly.

```
Java 8 date/time type `java.time.LocalDate` not supported by default: 
add Module "com.fasterxml.jackson.datatype:jackson-datatype-jsr310"
```